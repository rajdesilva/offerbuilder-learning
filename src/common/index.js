export { default as Loader } from "./Loader";
export { appConstants } from "./Constants";
export { appConstants_results } from "./Results_Constants";
export { AccessDenied } from "./AccessDenied";
export { UserConfirmation } from "./UserConfirmation";
