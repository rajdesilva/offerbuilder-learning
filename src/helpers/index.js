export * from './api-config';
export * from './history';
