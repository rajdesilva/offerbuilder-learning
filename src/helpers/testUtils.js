import React from "react";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { render as rtlRender } from "@testing-library/react";
import { getValue } from "./utility";
import rootReducers from "../redux/rootReducers";
import * as languages from "../langs";
import { IntlProvider } from "react-intl";
const initialReducerState = {};
window.getValue = getValue;

window.google = {
  maps: {
    Marker: class {},
    Map: class {},
    LatLngBounds: class {},
    places: {
      Autocomplete: class {},
      AutocompleteService: class {},
      PlacesServiceStatus: {
        INVALID_REQUEST: "INVALID_REQUEST",
        NOT_FOUND: "NOT_FOUND",
        OK: "OK",
        OVER_QUERY_LIMIT: "OVER_QUERY_LIMIT",
        REQUEST_DENIED: "REQUEST_DENIED",
        UNKNOWN_ERROR: "UNKNOWN_ERROR",
        ZERO_RESULTS: "ZERO_RESULTS",
      },
      PlacesAutocomplete: {
        INVALID_REQUEST: "INVALID_REQUEST",
        NOT_FOUND: "NOT_FOUND",
        OK: "OK",
        OVER_QUERY_LIMIT: "OVER_QUERY_LIMIT",
        REQUEST_DENIED: "REQUEST_DENIED",
        UNKNOWN_ERROR: "UNKNOWN_ERROR",
        ZERO_RESULTS: "ZERO_RESULTS",
      },
    },

    MarkerClusterer: class {},
    Geocoder: class {},
  },
};
Object.defineProperty(window, "matchMedia", {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

function render(
  ui,
  {
    initialState = initialReducerState,
    store = createStore(rootReducers, initialState),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <IntlProvider locale="en" messages={languages["en"]}>
          {children}
        </IntlProvider>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything
export * from "@testing-library/react";

// override render method
export { render };
