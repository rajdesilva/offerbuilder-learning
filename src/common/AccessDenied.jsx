import { Button } from "antd";
import React from "react";
import { FormattedMessage } from "react-intl";
import { useLocation } from "react-router-dom";
import { baseURL } from "../helpers";
import styles from './css/AccessDenied.module.less';

export function AccessDenied() {
  const query = new URLSearchParams(useLocation().search);
  const emailId = query.get("email");
  return (
    <div className={styles.accessDenied}>
      <div className={styles.screenTitle}>
        <FormattedMessage id="nemo.accessDenied" />
      </div>
      <div className={styles.deniedMsg}>
        <FormattedMessage
          id="nemo.notHavAccessToApplication"
          values={{
            email: emailId,
          }}
        />
        <br />
        <br />
        <FormattedMessage id="nemo.contactSystmeAdmin" />
      </div>
      <Button
        type="primary"
        data-testid="denied-logout-btn"
        className={styles.deniedLogoutBtn}
        onClick={() => (window.location.href = `${baseURL.API_URL_BASE}logout`)}
      >
        <FormattedMessage id="nemo.logout" />
      </Button>
    </div>
  );
}
