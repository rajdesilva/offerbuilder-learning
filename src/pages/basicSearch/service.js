import { browseSupplyActions, supplySearchActions } from "./actions";
import { store } from "./../../redux/store";
import { globalActions } from "../../redux/globalActions";
import { baseURL } from "../../helpers";
import {
  createSearchQueryURL,
  getSearchParamsToFetchMarketPrice,
} from "../../helpers/utility";
import { offerListSearchAndFilterActions } from "../offers/actions";

export const searchSupply = (routePostSearch, isEditFlow) => {
  store.dispatch({
    type: browseSupplyActions.BROWSE_SUPPLY_SEARCH_LOADING,
    payload: true,
  });
  store.dispatch({
    type: supplySearchActions.SET_INCLUDE_MARKET_PRICE_FLAG,
    payload: false,
  });

  window
    .fetchWrapper(`${baseURL.API_URL_BASE}search/` + createSearchQueryURL(routePostSearch))
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: browseSupplyActions.BROWSE_SUPPLY_SEARCH_SUCCESS,
          payload: window.getValue(response, "data.properties"),
          totalMatch: window.getValue(response, "data.totalMatchedProperties"),
          pageOffset: window.getValue(response, "data.pageOffset"),
          pageSize: window.getValue(response, "data.pageSize"),
        });
      } else {
        store.dispatch({
          type: browseSupplyActions.BROWSE_SUPPLY_SEARCH_FAILURE,
          payload: window.getValue(response, "error.apierror.message"),
        });
      }
    })
    .catch((error) => {
      store.dispatch({
        type: browseSupplyActions.BROWSE_SUPPLY_SEARCH_FAILURE,
        payload: error.toString(),
      });
      store.dispatch({
        type: globalActions.SET_ERROR,
        payload: error.toString(),
      });
    });
};

export const advancedSearch = (
  params,
  searchType,
  page,
  pageSize,
  sortBy,
  order,
  groupType
  ) => {
    return window.fetchWrapper(`${baseURL.API_URL_BASE}global/properties?address=${params.address}&city=${params.city}&countryCode=${params.countryCode}&distance=${params.distance}&hotelName=${params.hotelName}&latitude=${params.latitude}&longitude=${params.longitude}&zip=${params.zip}&pageOffset=${page}&pageSize=${pageSize}&propertyGroupType=${groupType}&sortCriteria=${sortBy}&sortOrder=${order}&globalSearchType=${searchType}`);

    //return window.fetchWrapper(`https://offerbuilder-api-dev.kognitiv.com/global/properties?distance=10&globalSearchType=BASIC&latitude=47.2816&longitude=15.48809&propertyGroupType=ALL`);
}

export const globalSearchResult =(params)=>{
  
  window
    .fetchWrapper(`/global/properties?address=${params.address}`)
    .then((response) => {
        console.log(response)
    })
    .catch((error) => {
      store.dispatch({
        type: supplySearchActions.SEARCH_CHANNELS_FAILURE,
        payload: [],
      });
      return [];
    });
}

export const getChannels = () => {
  store.dispatch({
    type: supplySearchActions.SEARCH_CHANNELS_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(`${baseURL.API_URL_BASE}portals`)
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: supplySearchActions.SEARCH_CHANNELS_SUCCESS,
          payload: window.getValue(response, "data.result"),
        });
      } else {
        store.dispatch({
          type: supplySearchActions.SEARCH_CHANNELS_FAILURE,
          payload: [],
        });
      }
    })
    .catch((error) => {
      store.dispatch({
        type: supplySearchActions.SEARCH_CHANNELS_FAILURE,
        payload: [],
      });
      return [];
    });
};

export const getBrands = async () => {
  store.dispatch({
    type: supplySearchActions.SEARCH_BRANDS_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(`${baseURL.API_URL_BASE}sources`)
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: supplySearchActions.SEARCH_BRANDS_SUCCESS,
          payload: window.getValue(response, "data.brands"),
        });
        store.dispatch({
          type: offerListSearchAndFilterActions.UPDATE_OFFER_SELECTORS,
          payload: window.getValue(response, "data.brands"),
        });
      } else {
        store.dispatch({
          type: supplySearchActions.SEARCH_BRANDS_FAILURE,
          payload: [],
        });
      }
      return window.getValue(response, "data.brands");
    })
    .catch((error) => {
      store.dispatch({
        type: supplySearchActions.SEARCH_BRANDS_FAILURE,
        payload: [],
      });
      return [];
    });
};

export const getCurrencies = async () => {
  store.dispatch({
    type: supplySearchActions.FETCH_CURRENCIES_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(`${baseURL.API_URL_BASE}currencies`)
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: supplySearchActions.FETCH_CURRENCIES_SUCCESS,
          payload: window.getValue(response, "data.currencies"),
        });
      } else {
        store.dispatch({
          type: supplySearchActions.FETCH_CURRENCIES_FAILURE,
          payload: [],
        });
      }
    })
    .catch((error) => {
      store.dispatch({
        type: supplySearchActions.FETCH_CURRENCIES_FAILURE,
        payload: [],
      });
      return [];
    });
};

export const getMarketPricesForListOfProperties = () => {
  store.dispatch({
    type: supplySearchActions.FETCH_MARKET_PRICE_LOADING,
    payload: true,
  });

  window
    .fetchWrapper(
      `${baseURL.API_URL_BASE}search/marketprice/` +
        getSearchParamsToFetchMarketPrice()
    )
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: supplySearchActions.FETCH_MARKET_PRICE_SUCCESS,
          payload: window.getValue(response, "data"),
        });
      } else {
        store.dispatch({
          type: supplySearchActions.FETCH_MARKET_PRICE_FAILURE,
          payload: [],
        });
      }
    })
    .catch((error) => {
      store.dispatch({
        type: supplySearchActions.FETCH_MARKET_PRICE_FAILURE,
        payload: [],
      });
      return [];
    });
};
