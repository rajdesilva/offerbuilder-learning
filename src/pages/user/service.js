import { baseURL } from "../../helpers";
import { userActions } from "./actions/userActions";
import { store } from "../../redux/store";

export const fetchUserInfo = () => {
  const url = `${baseURL.API_URL_BASE}user/`;
  store.dispatch({
    type: userActions.FETCH_USER_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(url)
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: userActions.FETCH_USER_SUCCESS,
          payload: window.getValue(response, "data"),
        });
      } else {
        store.dispatch({
          type: userActions.FETCH_USER_FAILURE,
          payload: false,
        });
      }
    })
    .catch(() => {
      store.dispatch({
        type: userActions.FETCH_USER_FAILURE,
        payload: false,
      });
    });
};

export const getUsersList = () => {
  store.dispatch({
    type: userActions.FETCH_USER_MANAGEMENT_LIST_LOADING,
    payload: true,
  });
  window
    .fetchWrapper(
      `${baseURL.API_URL_BASE}users/`
    )
    .then((response) => {
      if (response.success) {
        store.dispatch({
          type: userActions.FETCH_USER_MANAGEMENT_LIST_SUCCESS,
          payload: window.getValue(response, "data"),
        });
      } else {
        store.dispatch({
          type: userActions.FETCH_USER_MANAGEMENT_LIST_FAILURE,
          payload: [],
        });
      }
    })
    .catch((error) => {
      store.dispatch({
        type: userActions.FETCH_USER_MANAGEMENT_LIST_FAILURE,
        payload: [],
      });
    });
};

export const addNewUser = (newUserInfo) => {
  const url = `${baseURL.API_URL_BASE}users/user`;
  return window
    .fetchWrapper(url, {
      method: "POST",
      body: JSON.stringify(newUserInfo),
    })
    .then((response) => response)
    .catch((error) => {
      return error;
    });
};

export const deleteUser = (email, userRole) => {
  const url = `${baseURL.API_URL_BASE}users/user/${email}?roleNames=${userRole}`;
  return window
    .fetchWrapper(url, {
      method: "DELETE",
    })
    .then((response) => response)
    .catch((error) => {
      return error;
    });
};
