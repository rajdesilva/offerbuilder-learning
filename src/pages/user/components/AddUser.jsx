import React, { Fragment, useState } from "react";
import { Row, Col, Input, Radio, Switch, Form, message } from "antd";
import { FormattedMessage, useIntl } from "react-intl";
import AddUserHeader from "./AddUserHeader";
import TextArea from "antd/lib/input/TextArea";
import { appConstants, Loader } from "../../../common";
import { addNewUser } from "../service";
import { history } from "../../../helpers";
import styles from "./css/AddUser.module.less";

export default function AddUser() {
  const intl = useIntl();
  const [form] = Form.useForm();
  const [loading, setLoading] = useState(false);

  const checkFormFields = () => {
    setLoading(true);
    form
      .validateFields()
      .then((formValues) => {
        setLoading(true);

        addNewUser({
          email: formValues.email,
          name: formValues.name,
          userRole: formValues.userRole,
          customText: formValues.customText,
          inviteEmail: formValues.inviteEmail,
        }).then((response) => {
          setLoading(false);
          if (window.getValue(response, "success")) {
            message.success(
              intl.formatMessage({
                id: "nemo.createUserSuccess",
              })
            );
            history.push("/user-management/list");
          } else {
            if (window.getValue(response, "error.message")) {
              message.error(window.getValue(response, "error.message"));
            } else {
              message.error(
                intl.formatMessage({
                  id: "nemo.failedToCreateUser",
                })
              );
            }
          }
        });
      })
      .catch((error) => {
        setLoading(false);
        console.log("form error", error);
        message.error(error.toString());
      });
  };

  return (
    <React.Fragment>
      <AddUserHeader
        checkFormFields={checkFormFields}
        resetForm={() => form.resetFields()}
      />
      <Loader spinning={loading}>
        <Form
          form={form}
          scrollToFirstError
          initialValues={{
            inviteEmail: true,
            userRole: appConstants.USER_ROLE.ADMIN,
          }}
          className={styles["add-user-form"]}
          data-testid="add-user-management-form"
        >
          <Row gutter={[8, 8]} className={styles["form-row"]}>
            <Col span={6} className={styles["col-1-text"]}>
              <FormattedMessage id="nemo.name" />
            </Col>
            <Col span={6}>
              <Form.Item
                name="name"
                data-testid="add-user-name-form-item"
                rules={[
                  {
                    required: true,
                    message: <FormattedMessage id="nemo.pleaseEnterName" />,
                  },
                  {
                    min: 5,
                    message: (
                      <FormattedMessage id="nemo.nameMustBeMinimum5Characters." />
                    ),
                  },
                  {
                    max: 30,
                    message: (
                      <FormattedMessage id="nemo.nameCanBe30CharactersLong" />
                    ),
                  },
                ]}
              >
                <Input
                  data-testid="add-user-name"
                  maxLength={30}
                  placeholder={intl.formatMessage({
                    id: "nemo.userName",
                  })}
                />
              </Form.Item>
            </Col>
          </Row>

          <Row gutter={[8, 8]} className={styles["form-row"]}>
            <Col span={6} className={styles["col-1-text"]}>
              <FormattedMessage id="nemo.email" />
            </Col>
            <Col span={6}>
              <Form.Item
                name="email"
                data-testid="add-user-email-form-item"
                rules={[
                  {
                    required: true,
                    pattern:
                      /^[a-zA-Z-0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,10}$/,
                    message: (
                      <FormattedMessage id="nemo.pleaseEnterValidEmail" />
                    ),
                  },
                ]}
              >
                <Input
                  data-testid="add-user-email"
                  type="email"
                  placeholder={intl.formatMessage({
                    id: "nemo.emailAddress",
                  })}
                />
              </Form.Item>
            </Col>
          </Row>
          <Row gutter={[8, 8]} className={styles["form-row"]}>
            <Col span={6} className={styles["col-1-text"]}>
              <FormattedMessage id="nemo.userRole" />
            </Col>
            <Col span={6}>
              <Form.Item
                name="userRole"
                data-testid="add-user-role-form-item"
                initialValue="admin"
              >
                <Radio.Group>
                  <Radio
                    className={styles.radioBtn}
                    data-testid="admin-radio-btn"
                    value={appConstants.USER_ROLE.ADMIN}
                  >
                    <span className={styles.radioTextSpan}>
                      <b>
                        <FormattedMessage id="nemo.admin" />
                      </b>
                      <FormattedMessage id="nemo.adminCanDoThis" />
                    </span>
                  </Radio>
                  <Radio
                    className={styles.radioBtn}
                    data-testid="editor-radio-btn"
                    value={appConstants.USER_ROLE.EDITOR}
                  >
                    <span className={styles.radioTextSpan}>
                      <b>
                        <FormattedMessage id="nemo.editor" />
                      </b>
                      <FormattedMessage id="nemo.editorCanDoThis" />
                    </span>
                  </Radio>
                  <Radio
                    className={styles.radioBtn}
                    data-testid="viewer-radio-btn"
                    value={appConstants.USER_ROLE.VIEWER}
                  >
                    <span className={styles.radioTextSpan}>
                      <b>
                        <FormattedMessage id="nemo.viewer" />
                      </b>
                      <FormattedMessage id="nemo.viewerCanDoThis" />
                    </span>
                  </Radio>
                </Radio.Group>
              </Form.Item>
            </Col>
          </Row>
          <Row gutter={[8, 8]} className={styles["form-row"]}>
            <Col span={6} className={styles["col-1-text"]}>
              <FormattedMessage id="nemo.sendAnInviteEmail" />
            </Col>
            <Col span={6}>
              <Form.Item
                name="inviteEmail"
                valuePropName="checked"
                data-testid="add-user-send-invite-form-item"
              >
                <Switch data-testid="add-user-send-invite-switch" />
              </Form.Item>
            </Col>
          </Row>
          <Form.Item
            noStyle
            shouldUpdate={(prevValues, currentValues) =>
              prevValues.inviteEmail !== currentValues.inviteEmail
            }
          >
            {({ getFieldValue }) => {
              return getFieldValue("inviteEmail") ? (
                <Row gutter={[8, 8]} className={styles["form-row"]}>
                  <Col span={6} className={styles["col-1-text"]}>
                    <FormattedMessage id="nemo.emailContent" />
                  </Col>
                  <Col span={14}>
                    <Row>
                      <FormattedMessage id="nemo.customTextAddedAndWillBeListedBelow" />
                    </Row>
                    <Form.Item
                      name="customText"
                      className={styles.textFormItem}
                      data-testid="add-user-email-content-form-item"
                    >
                      <TextArea
                        data-testid="add-user-email-invite-textarea"
                        rows={6}
                        allowClear
                        placeholder={intl.formatMessage({
                          id: "nemo.textareaPlaceholder",
                        })}
                      />
                    </Form.Item>
                  </Col>
                </Row>
              ) : (
                <Fragment />
              );
            }}
          </Form.Item>
        </Form>
      </Loader>
    </React.Fragment>
  );
}
