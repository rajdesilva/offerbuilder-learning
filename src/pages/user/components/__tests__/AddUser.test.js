import React from "react";
import { createStore } from "redux";

import { render, fireEvent, cleanup } from "../../../../helpers/testUtils";
import { MemoryRouter } from "react-router-dom";
import AddUser from "../AddUser";
import * as service from "./../../service";

import { act } from "react-dom/test-utils";
import { history } from "../../../../helpers";
import { appConstants } from "../../../../common";

window.fetchWrapper = window.fetchWrapper = jest.fn().mockResolvedValue({
  success: true,
});

jest.mock("./../../service");
const tempStore = {
  usermanagement: {
    loading: false,
    userList: [],
  },
};

afterEach(cleanup);

describe("AddUser component test", () => {
  test("create AddUser components snapshot", () => {
    const store = createStore(() => ({ ...tempStore }));
    store.dispatch = jest.fn();
    const { container } = render(
      <MemoryRouter initialEntries={["/user-management/add-user"]}>
        <AddUser />
      </MemoryRouter>,
      {
        store,
      }
    );

    expect(container).toMatchSnapshot();
  });
  test("AddUser components- click on create user button and check validations messages", async () => {
    const store = createStore(() => ({ ...tempStore }));
    store.dispatch = jest.fn();
    const { findByText, getByTestId } = render(
      <MemoryRouter initialEntries={["/user-management/add-user"]}>
        <AddUser />
      </MemoryRouter>,
      {
        store,
      }
    );
    fireEvent.click(getByTestId("user-management-add-user-create-btn"));
    const items = await findByText(/Please Enter Name/);
    expect(items).toHaveTextContent("Please Enter Name");
    const items1 = await findByText(/Please Enter Valid Email/);
    expect(items1).toHaveTextContent("Please Enter Valid Email");
  });
  test("AddUser components- click on Cancel button", () => {
    act(() => {
      const store = createStore(() => ({ ...tempStore }));
      store.dispatch = jest.fn();
      const { getByTestId } = render(
        <MemoryRouter initialEntries={["/user-management/add-user"]}>
          <AddUser />
        </MemoryRouter>,
        {
          store,
        }
      );
      const cancelBtn = getByTestId("user-management-add-user-cancel-btn");
      fireEvent.click(cancelBtn);
      expect(cancelBtn).toBeTruthy();
      expect(history.location.pathname).toBe("/user-management/list");
    });
  });
  test("AddUser components- click on Create button for checking invalid entered values", async () => {
    const store = createStore(() => ({ ...tempStore }));
    store.dispatch = jest.fn();
    const { getByTestId, findByText, getByPlaceholderText } = render(
      <MemoryRouter initialEntries={["/user-management/add-user"]}>
        <AddUser />
      </MemoryRouter>,
      {
        store,
      }
    );
    //check for minimum limit chars
    let input = getByPlaceholderText(/User Name/);
    fireEvent.change(input, { target: { value: "Swa" } }); // 3 chars
    expect(input.value).toBe("Swa");
    fireEvent.click(getByTestId("user-management-add-user-create-btn"));
    const items = await findByText(/Name Must Be Minimum 5 Characters/);
    expect(items).toHaveTextContent("Name Must Be Minimum 5 Characters");

    //check for email entered valid or not
    let emailInput = getByPlaceholderText(/Email Address/);
    fireEvent.change(emailInput, { target: { value: "test@test" } }); // invalid email
    expect(emailInput.value).toBe("test@test");
    fireEvent.click(getByTestId("user-management-add-user-create-btn"));
    const items1 = await findByText(/Please Enter Valid Email/);
    expect(items1).toHaveTextContent("Please Enter Valid Email");
  });

  test("AddUser components- click on Create button for checking valid entered values", async () => {
    const store = createStore(() => ({ ...tempStore }));
    store.dispatch = jest.fn();
    const { getByTestId, queryByText, getByPlaceholderText } = render(
      <MemoryRouter initialEntries={["/user-management/add-user"]}>
        <AddUser />
      </MemoryRouter>,
      {
        store,
      }
    );
    //check for minimum limit chars
    let input = getByPlaceholderText(/User Name/);
    fireEvent.change(input, { target: { value: "Swapnil" } }); // 3 chars
    expect(input.value).toBe("Swapnil");

    fireEvent.click(getByTestId("user-management-add-user-create-btn"));
    const items = await queryByText(/Name Must Be Minimum 5 Characters/);
    expect(items).not.toBeInTheDocument();

    //check for email entered valid or not
    let emailInput = getByPlaceholderText(/Email Address/);
    fireEvent.change(emailInput, { target: { value: "test@test.com" } }); // invalid email
    expect(emailInput.value).toBe("test@test.com");
    fireEvent.click(getByTestId("user-management-add-user-create-btn"));
    const items1 = await queryByText(/Please Enter Valid Email/);
    expect(items1).not.toBeInTheDocument();
  });

  test("AddUser components- change radio button for changing roles values", async () => {
    const store = createStore(() => ({ ...tempStore }));
    store.dispatch = jest.fn();
    const { getByTestId } = render(
      <MemoryRouter initialEntries={["/user-management/add-user"]}>
        <AddUser />
      </MemoryRouter>,
      {
        store,
      }
    );
    const radio = getByTestId("admin-radio-btn");
    fireEvent.change(radio, {
      target: { value: appConstants.USER_ROLE.EDITOR },
    });
    expect(radio.value).toBe(appConstants.USER_ROLE.EDITOR);
  });

  test("AddUser components- Create clicked, All fields are valid and API is called and returned successful response", async () => {
    const tempStore1 = {
      usermanagement: {
        loading: false,
        userList: [],
      },
    };
    const store = createStore(() => ({ ...tempStore1 }));
    const { getByTestId, getByPlaceholderText } = render(
      <MemoryRouter initialEntries={["/user-management/add-user"]}>
        <AddUser />
      </MemoryRouter>,
      {
        store,
      }
    );
    // mocking post api when submit button is clicked
    service.addNewUser.mockImplementation((data) => {
      return Promise.resolve({
        success: true,
      });
    });

    const formData = {
      name: "Swapnil",
      email: "swapnil.deshmukh@kognitiv.com",
      customText: "Test",
      inviteEmail: true,
      userRole: appConstants.USER_ROLE.EDITOR,
    };

    let input = getByPlaceholderText(/Email Address/);
    fireEvent.change(input, { target: { value: formData.email } });
    input = getByPlaceholderText(/User Name/);
    fireEvent.change(input, { target: { value: formData.name } });
    input = getByPlaceholderText(/Text are a Placeholder/);
    fireEvent.change(input, { target: { value: formData.customText } });
    const radio = getByTestId("admin-radio-btn");
    fireEvent.change(radio, { target: { value: formData.userRole } });
    const switchBtn = getByTestId("add-user-send-invite-switch");
    fireEvent.change(switchBtn, { target: { value: formData.inviteEmail } });
    expect(switchBtn.value).toBe("true");
    await act(async () => {
      await fireEvent.click(getByTestId("user-management-add-user-create-btn"));
    });
    expect(history.location.pathname).toBe("/user-management/list");
  });
});
