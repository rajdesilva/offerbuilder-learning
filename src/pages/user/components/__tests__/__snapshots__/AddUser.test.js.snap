// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddUser component test create AddUser components snapshot 1`] = `
<div>
  <div
    class="ant-row ant-row-space-between"
  >
    <div
      class="ant-col"
    >
      <h2>
        Add User
      </h2>
    </div>
    <div
      class="ant-col"
    >
      <button
        class="ant-btn ant-btn-primary ant-btn-lg ant-btn-background-ghost"
        data-testid="user-management-add-user-cancel-btn"
        type="button"
      >
        Cancel
      </button>
      <button
        class="ant-btn ant-btn-primary ant-btn-lg"
        data-testid="user-management-add-user-create-btn"
        type="button"
      >
        Create
      </button>
    </div>
  </div>
  <div
    class="ant-divider ant-divider-horizontal"
    role="separator"
  />
  <div
    class="ant-spin-nested-loading"
    data-testid="loader"
  >
    <div
      class="ant-spin-container"
    >
      <form
        class="ant-form ant-form-horizontal"
        data-testid="add-user-management-form"
      >
        <div
          class="ant-row"
          style="margin: -4px -4px 4px -4px;"
        >
          <div
            class="ant-col ant-col-6"
            style="padding: 4px 4px 4px 4px;"
          >
            Name
          </div>
          <div
            class="ant-col ant-col-6"
            style="padding: 4px 4px 4px 4px;"
          >
            <div
              class="ant-row ant-form-item"
              data-testid="add-user-name-form-item"
            >
              <div
                class="ant-col ant-form-item-control"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <input
                      class="ant-input"
                      data-testid="add-user-name"
                      id="name"
                      maxlength="30"
                      placeholder="User Name"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row"
          style="margin: -4px -4px 4px -4px;"
        >
          <div
            class="ant-col ant-col-6"
            style="padding: 4px 4px 4px 4px;"
          >
            Email
          </div>
          <div
            class="ant-col ant-col-6"
            style="padding: 4px 4px 4px 4px;"
          >
            <div
              class="ant-row ant-form-item"
              data-testid="add-user-email-form-item"
            >
              <div
                class="ant-col ant-form-item-control"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <input
                      class="ant-input"
                      data-testid="add-user-email"
                      id="email"
                      placeholder="Email Address"
                      type="email"
                      value=""
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row"
          style="margin: -4px -4px 4px -4px;"
        >
          <div
            class="ant-col ant-col-6"
            style="padding: 4px 4px 4px 4px;"
          >
            User Role
          </div>
          <div
            class="ant-col ant-col-6"
            style="padding: 4px 4px 4px 4px;"
          >
            <div
              class="ant-row ant-form-item"
              data-testid="add-user-role-form-item"
            >
              <div
                class="ant-col ant-form-item-control"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <div
                      class="ant-radio-group ant-radio-group-outline"
                      id="userRole"
                    >
                      <label
                        class="ant-radio-wrapper ant-radio-wrapper-checked"
                      >
                        <span
                          class="ant-radio ant-radio-checked"
                        >
                          <input
                            checked=""
                            class="ant-radio-input"
                            data-testid="admin-radio-btn"
                            type="radio"
                            value="offerbuilder.admin"
                          />
                          <span
                            class="ant-radio-inner"
                          />
                        </span>
                        <span>
                          <span>
                            <b>
                              Admin
                            </b>
                            Search supply, Create and manage offers, Add users
                          </span>
                        </span>
                      </label>
                      <label
                        class="ant-radio-wrapper"
                      >
                        <span
                          class="ant-radio"
                        >
                          <input
                            class="ant-radio-input"
                            data-testid="editor-radio-btn"
                            type="radio"
                            value="offerbuilder.editor"
                          />
                          <span
                            class="ant-radio-inner"
                          />
                        </span>
                        <span>
                          <span>
                            <b>
                              Editor
                            </b>
                            Search supply, Create and manage offers
                          </span>
                        </span>
                      </label>
                      <label
                        class="ant-radio-wrapper"
                      >
                        <span
                          class="ant-radio"
                        >
                          <input
                            class="ant-radio-input"
                            data-testid="viewer-radio-btn"
                            type="radio"
                            value="offerbuilder.viewer"
                          />
                          <span
                            class="ant-radio-inner"
                          />
                        </span>
                        <span>
                          <span>
                            <b>
                              Viewer
                            </b>
                            Search supply, only view offers
                          </span>
                        </span>
                      </label>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row"
          style="margin: -4px -4px 4px -4px;"
        >
          <div
            class="ant-col ant-col-6"
            style="padding: 4px 4px 4px 4px;"
          >
            Send An Invite Email
          </div>
          <div
            class="ant-col ant-col-6"
            style="padding: 4px 4px 4px 4px;"
          >
            <div
              class="ant-row ant-form-item"
              data-testid="add-user-send-invite-form-item"
            >
              <div
                class="ant-col ant-form-item-control"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <button
                      aria-checked="true"
                      class="ant-switch ant-switch-checked"
                      data-testid="add-user-send-invite-switch"
                      id="inviteEmail"
                      role="switch"
                      type="button"
                    >
                      <div
                        class="ant-switch-handle"
                      />
                      <span
                        class="ant-switch-inner"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="ant-row"
          style="margin: -4px -4px 4px -4px;"
        >
          <div
            class="ant-col ant-col-6"
            style="padding: 4px 4px 4px 4px;"
          >
            Email Content
          </div>
          <div
            class="ant-col ant-col-14"
            style="padding: 4px 4px 4px 4px;"
          >
            <div
              class="ant-row"
            >
              Custom Text Added And Will Be Listed Below
            </div>
            <div
              class="ant-row ant-form-item"
              data-testid="add-user-email-content-form-item"
            >
              <div
                class="ant-col ant-form-item-control"
              >
                <div
                  class="ant-form-item-control-input"
                >
                  <div
                    class="ant-form-item-control-input-content"
                  >
                    <span
                      class="ant-input-affix-wrapper ant-input-affix-wrapper-textarea-with-clear-btn"
                    >
                      <textarea
                        class="ant-input"
                        data-testid="add-user-email-invite-textarea"
                        id="customText"
                        placeholder="Text are a Placeholder"
                        rows="6"
                      />
                      <span
                        aria-label="close-circle"
                        class="anticon anticon-close-circle ant-input-clear-icon-hidden ant-input-clear-icon"
                        role="button"
                        tabindex="-1"
                      >
                        <svg
                          aria-hidden="true"
                          data-icon="close-circle"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="64 64 896 896"
                          width="1em"
                        >
                          <path
                            d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm165.4 618.2l-66-.3L512 563.4l-99.3 118.4-66.1.3c-4.4 0-8-3.5-8-8 0-1.9.7-3.7 1.9-5.2l130.1-155L340.5 359a8.32 8.32 0 01-1.9-5.2c0-4.4 3.6-8 8-8l66.1.3L512 464.6l99.3-118.4 66-.3c4.4 0 8 3.5 8 8 0 1.9-.7 3.7-1.9 5.2L553.5 514l130 155c1.2 1.5 1.9 3.3 1.9 5.2 0 4.4-3.6 8-8 8z"
                          />
                        </svg>
                      </span>
                    </span>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
